lib/ansible/modules/cloud/scaleway/scaleway_sshkey.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+#
+# Scaleway SSH keys management module
+#
+# Copyright (C) 2018 Online SAS.
+# https://www.scaleway.com
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: scaleway_sshkey
+short_description: Scaleway SSH keys management module
+version_added: "2.6"
+author: Remy Leone (@sieben)
+description:
+    - "This module manages SSH keys on Scaleway account."
+
+options:
+  state:
+    description:
+     - Indicate desired state of the SSH key.
+    default: present
+    choices: ['present', 'absent']
+  ssh_pub_key:
+    description:
+     - The Public SSH key to add.
+    required: false
+    default: None
+  oauth_token:
+    description:
+     - Scaleway OAuth token.
+    required: true
+    version_added: 2.4
+  timeout:
+    description:
+    - Timeout for API calls
+    required: false
+    default: 30
+
+requirements:
+  - "python >= 2.6"
+'''
+
+EXAMPLES = '''
+- name: "Add SSH key"
+  scaleway_sshkey:
+    ssh_pub_key: "ssh-rsa AAAA..."
+    state: "Present"
+
+- name: "Delete SSH key"
+  scaleway_sshkey:
+    ssh_pub_key: "ssh-rsa AAAA..."
+    state: "absent"
+'''
+
+RETURN = '''
+# Scaleway API info https://developer.scaleway.com/
+data:
+    description: This is only present when C(state=present)
+    returned: when C(state=present)
+    type: dict
+    sample: {
+        "ssh_public_keys": [
+            {"key": "ssh-rsa AAAA...."}
+        ]
+    }
+'''
+
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.urls import fetch_url
+
+
+class Response(object):
+
+    def __init__(self, resp, info):
+        self.body = None
+        if resp:
+            self.body = resp.read()
+        self.info = info
+
+    @property
+    def json(self):
+        if not self.body:
+            if "body" in self.info:
+                return json.loads(self.info["body"])
+            return None
+        try:
+            return json.loads(self.body)
+        except ValueError:
+            return None
+
+    @property
+    def status_code(self):
+        return self.info["status"]
+
+
+class AccountAPI(object):
+
+    def __init__(self, module, headers):
+        self.module = module
+        self.headers = headers
+        self.base_url = 'https://account.scaleway.com'
```

Comment:
Would it be useful to be able to set this URL ?

Reply:
Not really because this URL does not change.

Reply:
I'd lean towards making it an optional module arg as well. Mostly for testing purposes but also to support if the url does change users could handle it with just playbook updates instead of requiring a new module version.